src/utils/set-timeout-context.js

```diff
@@ -9,5 +9,10 @@ import bindFn from './bind-fn';
  * @returns {number}
  */
 export default function setTimeoutContext(fn, timeout, context) {
-  return setTimeout(bindFn(fn, context), timeout);
```

Comment:
I'd prefer to use `resolve().then` to schedule a microtask (high priority with immediate flush) to ensure we preserve async semantics.